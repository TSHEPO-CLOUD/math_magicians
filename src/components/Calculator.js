
import React from 'react';
import './Calculator.css';
import calculate from '../logic/calculator';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (buttonName) => {
    this.setState((state) => calculate(state, buttonName));
  };

  displayResult = () => {
    const { total, next, operation } = this.state;
    const display = (total || '') + (operation || '') + (next || '');

    return display || '0';
  };