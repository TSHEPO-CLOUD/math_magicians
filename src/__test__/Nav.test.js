import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import Calculator from "../components/Calculator";

describe("App", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Calculator />);

    expect(tree).toMatchSnapshot();
  });

  it("Display 10.5", () => {
    const { getByText } = render(<Calculator />);

    fireEvent.click(getByText("1"));
    fireEvent.click(getByText("6"));
    fireEvent.click(getByText("."));
    fireEvent.click(getByText("2"));
    fireEvent.click(getByText("รท"));
    fireEvent.click(getByText("2"));
    fireEvent.click(getByText("="));

    expect(getByText("8.1")).toBeInTheDocument();
  });
});
