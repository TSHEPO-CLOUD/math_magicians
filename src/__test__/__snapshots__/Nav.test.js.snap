// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div>
  <nav>
    <h1
      className="title"
    >
      Math Magicians
    </h1>
    <ul>
      <a
        href="/"
        onClick={[Function]}
      >
        <li
          style={
            Object {
              "color": "white",
            }
          }
        >
          Home
        </li>
      </a>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        <li
          className="middle"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Calculator
        </li>
      </a>
      <a
        href="/quote"
        onClick={[Function]}
      >
        <li
          style={
            Object {
              "color": "white",
            }
          }
        >
          Quote
        </li>
      </a>
    </ul>
  </nav>
</div>
`;
